src/openrct2-ui/windows/TrackList.cpp

```diff
@@ -337,7 +337,10 @@ static void window_track_list_mouseup(rct_window* w, rct_widgetindex widgetIndex
             // Keep the highlighted item selected
             if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
             {
-                w->selected_list_item = _filteredTrackIds[w->selected_list_item];
+                if (static_cast<ssize_t>(_filteredTrackIds.size() - 1) >= w->selected_list_item)
```

Comment:
Why not drop the `- 1` and use `>` instead instead of `>=`?

Reply:
…

Because it's late :sweat_smile: .

Reply:
I ended up improving the condition slightly, taking into account `w->selected_list_item` is sometimes set to -1 too.

https://github.com/OpenRCT2/OpenRCT2/blob/8351a0477d8d9c640e333b1342aa030eded4e209/src/openrct2-ui/windows/TitleEditor.cpp#L251

https://github.com/OpenRCT2/OpenRCT2/blob/8351a0477d8d9c640e333b1342aa030eded4e209/src/openrct2-ui/windows/ServerList.cpp#L149

https://github.com/OpenRCT2/OpenRCT2/blob/8351a0477d8d9c640e333b1342aa030eded4e209/src/openrct2/interface/Window_internal.h#L45